codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -625,4 +475,127 @@ private CharSequence commaSeparateEscapedValues(CharSequence currentValue, CharS
          */
         CharSequence escape(T value);
     }
+
+    private static final class CharSequenceToStringConverter implements TypeConverter<CharSequence, String> {
+
+        private static final CharSequenceToStringConverter INSTANCE = new CharSequenceToStringConverter();
+
+        @Override
+        public String toConvertedType(CharSequence value) {
+            return value.toString();
+        }
+
+        @Override
+        public CharSequence toUnconvertedType(String value) {
+            return value;
+        }
+    }
+
+    protected static class CharSequenceConverter implements ValueConverter<CharSequence> {
+
+        public static final CharSequenceConverter INSTANCE = new CharSequenceConverter();
+
+        @Override
+        public CharSequence convertObject(Object value) {
+            if (value instanceof CharSequence) {
+                return (CharSequence) value;
+            }
+            return value.toString();
+        }
+
+        @Override
+        public CharSequence convertInt(int value) {
+            return new AsciiString(String.valueOf(value));
+        }
+
+        @Override
+        public CharSequence convertLong(long value) {
+            return new AsciiString(String.valueOf(value));
+        }
+
+        @Override
+        public CharSequence convertDouble(double value) {
+            return new AsciiString(String.valueOf(value));
+        }
+
+        @Override
+        public CharSequence convertChar(char value) {
+            return String.valueOf(value);
+        }
+
+        @Override
+        public CharSequence convertBoolean(boolean value) {
+            return new AsciiString(String.valueOf(value));
+        }
+
+        @Override
+        public CharSequence convertFloat(float value) {
+            return new AsciiString(String.valueOf(value));
+        }
+
+        @Override
+        public boolean convertToBoolean(CharSequence value) {
+            return Boolean.parseBoolean(value.toString());
+        }
+
+        @Override
+        public CharSequence convertByte(byte value) {
+            return new AsciiString(String.valueOf(value));
+        }
+
+        @Override
+        public byte convertToByte(CharSequence value) {
+            return Byte.valueOf(value.toString());
+        }
+
+        @Override
+        public char convertToChar(CharSequence value) {
+            return value.charAt(0);
```

Comment:
You didn't introduce this (I prolly did :) ) but should we handle `value.isEmpty()`?